src/UnitTests/Core/Impl/Threading/UIThreadHelper.cs

```diff
@@ -56,12 +56,18 @@ public class UIThreadHelper : IMainThread {
         private ControlledTaskScheduler _taskScheduler;
         private AsyncLocal<BlockingLoop> _blockingLoop = new AsyncLocal<BlockingLoop>();
 
-        private UIThreadHelper() {}
+        private UIThreadHelper() { }
 
         public Thread Thread => _thread;
         public SynchronizationContext SyncContext => _syncContext;
         public ControlledTaskScheduler TaskScheduler => _taskScheduler;
 
+        #region IMainThread
+        int IMainThread.ThreadId => Thread.ManagedThreadId;
+        void IMainThread.Post(Action action, CancellationToken cancellationToken)
```

Comment:
No, that isn't what broken. The old code used `_blockingLoop`:
```cs
void IMainThread.Post(Action action, CancellationToken cancellationToken) {
    var bl = _blockingLoop.Value;
    if (bl != null) {
        bl.Post(action);
    } else {
        InvokeAsync(action, cancellationToken).DoNotWait();
    }
}
```

`_blockingLoop` is used to simulate CPS behavior when it blocks UI thread.